vm/systemSmartContracts/delegation.go

```diff
@@ -2055,22 +2132,74 @@ func (d *delegation) getDelegatorFundsData(args *vmcommon.ContractCallInput) vmc
 	return vmcommon.Ok
 }
 
-func (d *delegation) getDelegatorList(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	returnCode := d.checkArgumentsForGeneralViewFunc(args)
+func (d *delegation) getTotalCumulatedRewardsForUser(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	delegator, returnCode := d.checkArgumentsForUserViewFunc(args)
 	if returnCode != vmcommon.Ok {
 		return returnCode
 	}
 
-	delegationStatus, err := d.getDelegationStatus()
+	err := d.computeAndUpdateRewards(args.Arguments[0], delegator)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
-	for _, address := range delegationStatus.Delegators {
-		d.eei.Finish(address)
+	totalCumulatedRewards := big.NewInt(0).Add(delegator.TotalCumulatedRewards, delegator.UnClaimedRewards)
+	d.eei.Finish(totalCumulatedRewards.Bytes())
+
+	return vmcommon.Ok
+}
+
+func (d *delegation) setMetaData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	returnCode := d.checkOwnerCallValueGasAndDuplicates(args)
+	if returnCode != vmcommon.Ok {
+		return returnCode
+	}
+
+	if len(args.Arguments) != 3 {
+		d.eei.AddReturnMessage("needed 3 arguments")
```

Comment:
this can be easily tested

Reply:
common. checking number of arguments. it is not a prio. tests are needed in order to check some computation/logic not to improve coverage.

Reply:
It is high prio when doing merges. Remember! We do not write unit tests just for fun. I know this if should work well and there should be no bugs there but remember that others might maintain the code and do some merges in which they can make mistakes.

Reply:
tested this as well.